docs/technical-doc/storage-costs.rst

```diff
@@ -0,0 +1,29 @@
+========================
+Storage Costs
+========================
+
+Explanations
+------------
+
+In Massa the ledger is shared across all nodes of the network. We need to set a size limit to be able to run a node without
+having 100TB of storage which will cause a barrier for adoption and running nodes at home.
+We chose to limit the size to 1TB. Everyone can store data until the ledger reach 1TB. But how can we ensure this limit ?
+
+We chose to force users to lock coins when they claim storage space and so we created a correlation with storage and circulating coins.
+
+For each byte of storage you claim (for your address and balance, a key in your datastore, bytecode, ...) you need to lock coins. The coins are released when you release your space in the storage.
+
+The amount of coins you need to lock for one byte is 0.0002 MASS. This value has been choosen so that if all coins (500 000 000 MASS) are locked, we will reach the 1TB limit.
+
+As balances are stored as `varint <https://developers.google.com/protocol-buffers/docs/encoding#varints>`__ in the ledger their size can vary. To avoid difficulties and incomprehension we decided to use a fixed size for each balance. This size is 5 bytes and so cost 0.0002 * 5.
+Datastore keys also have a variable size and so we decided to use a fixed size of 10 bytes for calculating storage cost. The storage cost of a datastore key is (32 + 10) * 0.0002. 32 is the size of the address and 10 is the constant for datastore key.
```

Comment:
We could maybe condensate the address size, btreemap overhead size, separator size and balance size into a single LEDGER_ENTRY_BASE_SIZE that represents a "base size" for a ledger entry with empty datastore and bytecode